src/schemes/oauth2.ts

```diff
@@ -162,6 +162,7 @@ export default class Oauth2Scheme extends BaseScheme<typeof DEFAULTS> {
       // https://auth0.com/docs/protocols/oauth2/oauth-state
       state: _opts.state || nanoid(),
       code_challenge_method: this.options.codeChallengeMethod,
+      nonce: _opts.nonce,
```

Comment:
I think we should check if `response_type` is `token` instead of just make it settable for any response type :)

You can change the check of [line 176](https://github.com/gagandeep/auth-module/blob/dev/src/schemes/oauth2.ts#L176), then we don't need the [line 165](https://github.com/gagandeep/auth-module/blob/dev/src/schemes/oauth2.ts#L165)

Just need to change the `includes` to `token`:
```js
if (opts.response_type.includes('token')) {
```

What do you think?